---
title: "Emunkitug Tumor bulkRNAseq"
author: "Spencer Huggett"
date: "`r Sys.Date()`"
output: html_document
---


## Introduction

This analysis covers the preprocessing, differential expression, PCA, gene set enrichment, and visualization of bulk RNA-seq data. The goal is to provide a reproducible workflow for analyzing the data.

## Setup

Load required libraries. Make sure to install any packages that are not available in your environment.

```{r libraries, message=FALSE, warning=FALSE}
# Data analysis and visualization packages
library(DESeq2)
library(RSQLite)
library(data.table)
library(biomaRt)
library(factoextra)
library(fgsea)
library(ggsci)
library(ggpubr)
library(edgeR)
library(BiocParallel)
library(readxl)
library(plyr)
library(stringr)
library(dplyr)
library(ComplexHeatmap)
library(circlize)
library(ggfx)
library(ggrepel)
library(variancePartition)

makeGeneRankStat <- function(statistic, names) {
  generank <- statistic
  names(generank) <- names
  generank <- generank[which(!is.na(generank))]
  generank <- generank[which(!is.nan(generank))]
  generank <- generank[which(names(generank) != "")]
  generank[which(generank == "-Inf")] <- min(generank[which(generank != "-Inf")], na.rm = TRUE) - 
    (0.01 * min(generank[which(generank != "-Inf")], na.rm = TRUE))
  generank[which(generank == "Inf")] <- max(generank[which(generank != "Inf")], na.rm = TRUE) + 
    (0.01 * max(generank[which(generank != "Inf")], na.rm = TRUE))
  generank <- generank[order(abs(generank), decreasing = TRUE)]
  generank <- generank[!duplicated(names(generank))]
  generank <- generank[order(generank, decreasing = TRUE)]
  return(generank)
}


# Load gene info from biomaRt
human <- useMart("ensembl", dataset = "hsapiens_gene_ensembl")
humBM <- getBM(attributes = c("ensembl_gene_id", "hgnc_symbol", "description"), mart = human)
humBM$description <- gsub("\\s*\\[.*?\\]", "", humBM$description)

# Load metadata and raw count data
meta <- read.csv('~/HFB2003/Data/bulkRNAseq/2025/Metadata.2.26.25.csv')
dat <- as.data.frame(fread('~/HFB2003/Data/bulkRNAseq/2025/raw_mtx.tsv'))
colnames(dat) <- gsub('HFB200301_', '', colnames(dat))

# Extract gene IDs and clean data
gene_id <- dat$V1
dat <- dat[, which(colnames(dat) != 'V1')]
```





